pkg/command/output.go

```diff
@@ -86,13 +86,13 @@ func DumpJSONToSlice(data interface{}, jsonPath string) ([]byte, error) {
 		return nil, err
 	}
 
-	buf := new(bytes.Buffer)
-	if err := parser.Execute(buf, data); err != nil {
+	var sb strings.Builder
+	if err := parser.Execute(&sb, data); err != nil {
 		fmt.Fprintf(os.Stderr, "Couldn't parse jsonpath expression: '%s'\n", err)
 		return nil, err
 
 	}
-	return buf.Bytes(), nil
+	return []byte(sb.String()), nil
```

Comment:
It seems all users of this function convert the return value directly to a string before using it further. So we could even change the signature to:

```go
func DumpJSONToString(data interface{}, jsonPath string) (string, error)
```

and adjust the two callers accordingly. Might be better suited for a follow-up PR though.

Reply:
Done in https://github.com/cilium/cilium/pull/14041.